### PR TITLE
OWRank: Add ANOVA, Chi2 and Univariate Linear Regression

### PR DESCRIPTION
**Pull request by thocevar**
- Add existing feature scorers from sklearn to the widget (ANOVA, Chi2, Univariate Linear Regression)
- Add imputation to preprocessors for scores from sklearn (they fail with missing data)
- Attempt to score individual attributes in datasets with mixed attribute types

### CODE REVIEW

File: Orange/widgets/data/owrank.py [hunk 1]
```diff
@@ -531,10 +541,19 @@ def __lt__(self, other):
 
 
 class MySortProxyModel(QtGui.QSortFilterProxyModel):
+
+    @staticmethod
+    def comparable(val):
+        return val and float('-inf') < val < float('inf')
```
Comment (kernc) [inline Orange/widgets/data/owrank.py]: Does this sort nans below the rest?

Comment (kernc) [inline Orange/widgets/data/owrank.py]: Nevermind. :laughing:

Comment (thocevar) [inline Orange/widgets/data/owrank.py]: Nice catch.

**Comment by lanzagar** [comment]
Impute() preprocessor handles only missing values in features. If the class variable has them it still fails. Probably the RemoveNaNClasses preprocessor needs to be added as well?

**Pull request merged**